libs/pika/async_mpi/tests/unit/algorithm_transform_mpi.cpp

```diff
@@ -191,19 +188,18 @@ int pika_main()
                 }
                 catch (pika::exception const& e)
                 {
-                    PIKA_TEST_EQ(e.get_error(), pika::error::invalid_status);
-
-                    // Different MPI implementations print different error
-                    // messages. We handle MPICH and OpenMPI explicitly and
-                    // hope that for the rest it's enough to check that a
-                    // pika::exception was thrown with invalid_status.
-#if defined(MPICH)
-                    PIKA_TEST(std::string(e.what()).find(std::string("Invalid datatype")) !=
-                        std::string::npos);
-#elif defined(OPEN_MPI)
-                    PIKA_TEST(std::string(e.what()).find(std::string("MPI_ERR_TYPE")) !=
-                        std::string::npos);
-#endif
+                    // Different MPI implementations print different error messages.
+                    bool err_ok = (e.get_error() == pika::error::bad_function_call) ||
+                        (e.get_error() == pika::error::invalid_status);
+                    PIKA_TEST_MSG(err_ok, "Returned error code was not in expected list");
+                    std::vector<std::string> err_msgs = {
+                        "null datatype", "Invalid datatype", "MPI_ERR_TYPE"};
+                    bool msg_ok = false;
+                    for (const auto& msg : err_msgs)
+                    {
+                        msg_ok |= (std::string(e.what()).find(msg) != std::string::npos);
+                    }
+                    PIKA_TEST_MSG(msg_ok, "Error message did not contain expected string");
```

Comment:
I kind of liked being explicit about which MPI implementation produces what error, but understand that it might be annoying to keep up to date. Could you summarize how the exceptions/error messages changed from before? I currently don't see any explicit uses of `invalid_status` in the `async_mpi` module, but do see `bad_function_call`. Do we still expect `invalid_status` to be thrown, and if so, when?

Reply:
no. I just left the `invalid status` there cos that's what you had in your version, but I mistakenly thought we converted some mpi error status into the exception type, but in fact we always use `bad function call`. I'll remove it